plugin/core/src/main/java/fr/avianey/androidsvgdrawable/Colorizer.java

```diff
@@ -0,0 +1,178 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fr.avianey.androidsvgdrawable;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Describe the configuration for colorizing svg drawables:
+ * <dl>
+ *     <dt>Filename-matching</dt>
+ *     <dd>Regex against which to match filenames</dd>
+ *     <dt>Colorization</dt>
+ *     <dd>Input color (to be replaced)</dd>
+ *     <dd>Output color (replaces input color)</dd>
+ * </dl>
+ *
+ * For Example:
+ * <code>
+ *  [
+ *      {
+ *          "matches": ["ic_.*"],
+ *          "colors": [
+ *              {
+ *                  "from": "#FFFFFF",
+ *                  "to": "#FF0000"
+ *              }
+ *          ]
+ *      }
+ *  ]
+ * </code>
+ *
+ * will translate all white colors into red for
+ *  every file whose name starts with "ic_"
+ *
+ * @version 1
+ * @author dhleong
+ */
+public class Colorizer {
+
+    private static final Pattern FILL_STYLE = Pattern.compile("fill:([^;]+);");
+
+    static class ColorMap {
+        String from, to;
+    }
+
+    String[] matches;
+    ColorMap[] colors;
+
+    private List<Pattern> patterns;
+
+    public boolean matches(QualifiedResource qr) {
+        for (Pattern pattern : getPatterns()) {
+            if (pattern.matcher(qr.getName()).find())
+                return true;
+        }
+
+        return false;
+    }
+
+    public Document colorize(final QualifiedResource svg) {
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document doc = builder.parse(svg);
+
+            colorize(doc);
+            return doc;
+
+        } catch (ParserConfigurationException e) {
+            e.printStackTrace();
+        } catch (SAXException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private void colorize(final Node doc) {
+        final NamedNodeMap attrs = doc.getAttributes();
+        if (attrs != null) {
+            colorize(attrs);
+        }
+
+        final NodeList kids = doc.getChildNodes();
+        if (kids == null)
+            return;
+
+        final int len = kids.getLength();
+        for (int i=0; i < len; i++) {
+            final Node node = kids.item(i);
+            colorize(node);
+        }
+    }
+
+    private void colorize(final NamedNodeMap attrs) {
+        final Node fill = attrs.getNamedItem("fill");
```

Comment:
looks like https://xmlgraphics.apache.org/batik/javadoc/org/apache/batik/gvt/ShapeNode.html contains a ShapePainter which contains a Paint with the fill and stroke color, might be possible to tweak this